app/models/concerns/account_interactions.rb

```diff
@@ -82,21 +82,38 @@ def follow!(other_account, reblogs: nil)
     rel
   end
 
-  def block!(other_account)
-    block_relationships.find_or_create_by!(target_account: other_account)
+  def block!(other_account, note: nil)
+    transaction do
+      block = block_relationships.find_or_create_by!(target_account: other_account)
+      attach_note(block, note)
+      block
+    end
   end
```

Comment:
If there's a clean way to do this, it would probably be a good idea to revert these changes and move the note attachment into a separate mixin (`Glitch::AccountInteractions` maybe).  This will minimize the amount of code that could get caught up in a merge conflict.